share/eds/store.go

```diff
@@ -93,8 +91,13 @@ func (s *Store) Stop() error {
 // The square is verified on the Exchange level, and Put only stores the square, trusting it.
 // The resulting file stores all the shares and NMT Merkle Proofs of the EDS.
 // Additionally, the file gets indexed s.t. store.Blockstore can access them.
-func (s *Store) Put(ctx context.Context, root share.Root, square *rsmt2d.ExtendedDataSquare) error {
-	key := root.String()
+func (s *Store) Put(ctx context.Context, root DataHash, square *rsmt2d.ExtendedDataSquare) error {
+	// sanity check the root
+	if err := root.Validate(); err != nil {
+		return err
+	}
+
+	key := fmt.Sprintf("%X", root)
```

Comment:
make DataHash implement Stringer w the "%X"

Reply:
tru

Reply:
https://github.com/celestiaorg/celestia-node/pull/1439/commits/1d035d79e0b27f7964ee6971e45301d9fdb638b9